bzr.go

```diff
@@ -146,6 +147,40 @@ func (s *BzrRepo) Tags() ([]string, error) {
 	return tags, nil
 }
 
+// CurrentVersionsWithRevs returns a list of available branches and tags, and
+// their underlying revision identifiers. The list is guaranteed to be current.
+//
+// As the model we use for bzr interaction (and go get generally does as well)
+// involves interacting only with a bzr repository's default branch, we return
+// only tags here.
+func (s *BzrRepo) CurrentVersionsWithRevs() (versions []VersionInfo, localSynced bool, err error) {
+	var out []byte
+	// Update the local first
+	_, err = s.runFromDir("bzr", "pull")
+	if err != nil {
+		return
+	}
+	localSynced = true
```

Comment:
The purpose is to allow calling code to keep track of whether or not the local instance of a repository has the most up to date information, *without* triggering a potentially expensive network, or even filesystem, operation.

Which, actually, suggests what could be done here to get rid of that return value - if the repo objects were to include a simple `synced` boolean field, and we added an `IsSynced()` or `IsUpToDate()` method, then that method could return the value of the `synced` field. Methods like `Update()`, or the one being introduced here, would toggle that field on.

Obviously, this information would only be useful for short-lived processes - or at least, ones where it's acceptable that pulling updates from upstream only be done once. However, as long as that internal field isn't used to short-circuit calls to e.g. `Update()`, then there's no harm.